### PR TITLE
WIP reunvendor zlib

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -5,7 +5,8 @@ set -e
 if [ $(uname) = Darwin ] ; then
     # Needed to get 'union semun' definition used in drvrsmem.c:
     export CFLAGS="$CFLAGS -D_DARWIN_C_SOURCE"
-    export LDFLAGS="$LDFLAGS -Xlinker -reexport_library $PREFIX/lib/libz.dylib"
+    export LDFLAGS="$LDFLAGS -Wl,-reexport_library,$PREFIX/lib/libz.dylib"
+    export LIBS="$LIBS -Wl,-reexport_library,$PREFIX/lib/libz.dylib"
```
Comment (beckermr) [inline recipe/build.sh:9]: Thanks @isuruf! I was wondering what was going on here...

**Comment by beckermr** [comment]
frack - gdal is choking on test files not in the final package. Maybe we should just import gdal in the tests instead of running the full thing?

**Comment by isuruf** [comment]
https://github.com/conda-forge/gdal-feedstock/pull/385

**Comment by beckermr** [comment]
@conda-forge-admin restart ci

**Comment by beckermr** [comment]
ppc64le job fails with conflicts in the env.

**Comment by beckermr** [comment]
@isuruf given that we are checking for the symbol directly, can we punt on the downstream tests for now?

@brianv0 really needs this PR tonight and the gdal stuff is complicated.

**Comment by brianv0** [comment]
@conda-forge-admin restart ci

**Comment by brianv0** [comment]
With the gdal test cases merged, commit 28c54a1 passes locally on OS X for me, so I believe this PR does fix the issue, even if we haven't confirmed ppc64le due to env conflicts.

I'd strongly prefer reverting 5f1afa4 and 28c54a1 for now, then re-applying 28c54a1 at some point in the near future.

**Comment by beckermr** [comment]
Agreed. I’m on my phone now but I can do it later.

**Comment by beckermr** [comment]
We should scope the gdal test to OS X only.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -31,14 +30,15 @@ requirements:
...
-    # - zlib  # [not win]
+    - zlib  # [not win]
 
 test:
   commands:
     - test -f ${PREFIX}/lib/libcfitsio{{ SHLIB_EXT }}  # [not win]
     - if not exist %LIBRARY_LIB%\cfitsio.lib exit 1  # [win]
     - if not exist %LIBRARY_BIN%\cfitsio.dll exit 1  # [win]
+  downstreams:
+    - gdal
```
Comment (brianv0) [inline recipe/meta.yaml:41]:
```suggestion
    - gdal  # [osx]
```

Comment (beckermr) [reply recipe/meta.yaml]: We may need to decorate the line above too

**Comment by beckermr** [comment]
OS X tests failed in the right way. We need to uncomment out the lines and check again. Then we merge.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -31,14 +30,15 @@ requirements:
...
-    # - zlib  # [not win]
+    - zlib  # [not win]
 
 test:
   commands:
     - test -f ${PREFIX}/lib/libcfitsio{{ SHLIB_EXT }}  # [not win]
     - if not exist %LIBRARY_LIB%\cfitsio.lib exit 1  # [win]
     - if not exist %LIBRARY_BIN%\cfitsio.dll exit 1  # [win]
+  downstreams:
+    - gdal  # [osx]
```
Comment (isuruf) [inline recipe/meta.yaml:41]: Can you keep this for linux and win as well? The tests don't take that long and it helps keep gdal from failing which seems to be a common occurence. cc @ocefpaf.

Comment (beckermr) [reply recipe/meta.yaml:41]: We need to exclude ppc64le since this env does not solve. Otherwise it should be fine.

Comment (isuruf) [reply recipe/meta.yaml:41]: Ah, yeah. Seems to be a conda-build problem, https://github.com/conda/conda-build/issues/3908

Comment (isuruf) [reply recipe/meta.yaml:41]:
```suggestion
    - gdal  # [x86_64]
```

**Pull request merged**